clang/test/SemaSYCL/check-work-group-size-hint-device.cpp

```diff
@@ -17,7 +17,8 @@
 
 // Produce a conflicting attribute warning when the args are different.
 [[sycl::work_group_size_hint(4, 1, 1)]] void f3();    // expected-note {{previous attribute is here}}
-[[sycl::work_group_size_hint(1, 1, 32)]] void f3() {} // expected-warning {{attribute 'work_group_size_hint' is already applied with different arguments}}
```

Comment:
The additional warnings in all these tests makes the tests a bit messy IMO. I would just remove (if we already test that functionality elsewhere) or rewrite the incorrect tests to not generate this warning. But this might just be me nit-picking. @premanandrao @smanna12  what do you think?

Reply:
I think in this specific case, I need to remove 
`[[sycl::work_group_size_hint(1, 1, 32)]] void f3() {}`
because it can only be a host or device function, but not a kernel. As a result, applying this attribute generates a warning.

Reply:
I understand. My point was that these test cases are no longer really relevant in this context. For example in this case, if we already have other tests testing `work_group_size_hint` incorrectly applied on the kernel, we don't need to be testing it on device/host functions since we are not warning they are incorrect as per your new warning anyway.  It is not a big deal keeping these tests. It just makes the test cases a bit messy IMO.

I am ok submitting this PR as-is and cleaning up the tests in a separate PR if the other reviewers @premanandrao and @smanna12 want these tests cleaned up as well.

Reply:
I would like the tests cleaned up, but not necessarily as part of this PR.  Are you okay with that @elizabethandrews?

Reply:
>>I would like the tests cleaned up, but not necessarily as part of this PR

I have same opinion. Agree with @elizabethandrews and @premanandrao

Reply:
>>Are you okay with that @elizabethandrews?

Works for me. @DYNIO-INTEL please clean up the tests in a follow up PR

Reply:
It works for me too. I will clean up the tests in a follow up PR.